coral-hive/src/main/java/com/linkedin/coral/hive/hive2rel/functions/HiveFunctionResolver.java

```diff
@@ -43,14 +44,13 @@
 /**
  * Class to resolve hive function names in SQL to Function.
  */
-public class HiveFunctionResolver {
+public class HiveFunctionResolver extends FunctionResolver {
```

Comment:
Why could not not this be moved to coral-common?

Reply:
`HiveFunctionResolver` is only used for Coral-Hive and contains many Coral-Hive specific methods like Generic UDF dynamic registration. Therefore, I don't think we need this class for all the other xxx2rel modules, we need to have `xxxFunctionResolver` extending `FunctionResolver` for each xxx2rel module instead.

Reply:
@wmoustafa do you have a preference between common function resolver vs language-specific function resolvers?

Reply:
Not sure IIUC, I don't think we can use a common function resolver in `coral-common` to solve all the `language2rel` functions, given difference input languages may need difference methods in their function resolvers, for example, hive2rel needs dynamic Hive generic UDF registration while trino2rel doesn't need.

Reply:
I think aside from the dynamic resolution, rest should be language independent and go to coral-common?

> And in practice, given `HiveFunctionResolver` uses many classes in Coral-Hive like `VersionedSqlUserDefinedFunction`, `HiveRLikeOperator` and `HiveGenericUDFReturnTypeInference`, we need to move all of them for `HiveFunctionResolver`, which is not suitable in my opinion.

I think those are ultimately going to be Coral IR specific classes, and not Hive?

Reply:
@wmoustafa Rests are `resolveUnaryOperator` and `resolveBinaryOperator` which are only used in Hive's `ParseTreeBuilder`, not used in `trino2rel` because `trino2rel` has its own `ParseTreeBuilder`. So I would prefer to just keep `resolve` in `FunctionResolver` for now, we may add more common methods if there is in the future.